src/Entity/CreditMemo.php

```diff
@@ -50,101 +49,136 @@ class CreditMemo implements CreditMemoInterface
     /** @var ShopBillingDataInterface|null */
     protected $to;
 
-    public function __construct(
-        string $id,
-        string $number,
-        OrderInterface $order,
-        int $total,
-        string $currencyCode,
-        string $localeCode,
-        ChannelInterface $channel,
-        array $lineItems,
-        array $taxItems,
-        string $comment,
-        \DateTimeImmutable $issuedAt,
-        CustomerBillingDataInterface $from,
-        ?ShopBillingDataInterface $to
-    ) {
-        $this->id = $id;
-        $this->number = $number;
-        $this->order = $order;
-        $this->total = $total;
-        $this->currencyCode = $currencyCode;
-        $this->localeCode = $localeCode;
-        $this->channel = $channel;
-        $this->lineItems = new ArrayCollection($lineItems);
-        $this->taxItems = new ArrayCollection($taxItems);
-        $this->comment = $comment;
-        $this->issuedAt = $issuedAt;
-        $this->from = $from;
-        $this->to = $to;
-    }
-
     public function getId(): string
     {
         return $this->id;
     }
 
+    public function setId(string $id): void
+    {
+        $this->id = $id;
+    }
+
     public function getNumber(): string
     {
         return $this->number;
     }
 
+    public function setNumber(string $number): void
+    {
+        $this->number = $number;
+    }
+
     public function getOrder(): OrderInterface
     {
         return $this->order;
     }
 
+    public function setOrder(OrderInterface $order): void
+    {
+        $this->order = $order;
+    }
+
     public function getTotal(): int
     {
         return $this->total;
     }
 
+    public function setTotal(int $total): void
+    {
+        $this->total = $total;
+    }
+
     public function getCurrencyCode(): string
     {
         return $this->currencyCode;
     }
 
+    public function setCurrencyCode(string $currencyCode): void
+    {
+        $this->currencyCode = $currencyCode;
+    }
+
     public function getLocaleCode(): string
     {
         return $this->localeCode;
     }
 
+    public function setLocaleCode(string $localeCode): void
+    {
+        $this->localeCode = $localeCode;
+    }
+
     public function getChannel(): ChannelInterface
     {
         return $this->channel;
     }
 
+    public function setChannel(ChannelInterface $channel): void
+    {
+        $this->channel = $channel;
+    }
+
     public function getLineItems(): Collection
     {
         return $this->lineItems;
     }
 
+    public function setLineItems(Collection $lineItems): void
```

Comment:
Could it be replaced by the adder method?

Anyway, we should create an empty collection in the constructor as well - currently, it'll crash while flushing the changes if `setLineItems` and `setTaxItems` are not called.

Reply:
I've added creating collections in constructor, but I'm not convince to changing these setters (`setTaxItems()` and `setLineItems()`) to adders as these are collections that shouldn't have too many operations